toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -277,26 +304,48 @@ func (b *LiveOSIsoBuilder) updateGrubCfg(grubCfgFileName string, extraCommandLin
 //     b.artifacts.bootDirFiles
 func (b *LiveOSIsoBuilder) extractBootDirFiles(writeableRootfsDir string) error {
 
-	sourceBootDir := filepath.Join(writeableRootfsDir, "/boot")
 	b.artifacts.bootDirFiles = make(map[string]string)
 
-	var exclusions []*regexp.Regexp
 	// the following files will be re-created - no need to copy them only to
 	// have them overwritten.
+	var exclusions []*regexp.Regexp
+	//
+	// We will generate a new initrd later. So, we do not copy the initrd.img
+	// that comes in the input full disk image.
+	//
 	exclusions = append(exclusions, regexp.MustCompile(`/boot/initrd\.img.*`))
+	//
+	// On full disk images (generated by Mariner toolkit), there are two
+	// grub.cfg files:
+	// - <boot partition>/boot/grub2/grub.cfg:
+	//   - mounted at /boot/efi/boot/grub2/grub.cfg.
+	//   - empty except for redirection to the other grub.cfg.
+	// - <rootfs partition>/boot/grub2/grub.cfg:
+	//   - mounted at /boot/grub2/grub.cfg
+	//   - has the actual grub configuration.
+	//
+	// When creating an iso image out of a full disk image, we do not need the
+	// redirection mechanism, and hence we can do with only the full grub.cfg.
+	//
+	// To avoid confusion, we do not copy the redirection grub.cfg to the iso
+	// media.
+	//
+	exclusions = append(exclusions, regexp.MustCompile(`/boot/efi/boot/grub2/grub\.cfg`))
 
-	err := filepath.Walk(sourceBootDir, func(sourcePath string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
-		if info.IsDir() {
-			return nil
-		}
+	bootFolderFilePaths, err := file.EnumerateDirFiles(filepath.Join(writeableRootfsDir, "/boot"))
+	if err != nil {
+		return fmt.Errorf("failed to scan /boot folder:\n%w", err)
+	}
+
+	usingGrubNoPrefix := containsGrubNoPrefix(bootFolderFilePaths)
+
+	for _, sourcePath := range bootFolderFilePaths {
 
 		for _, exclusion := range exclusions {
 			match := exclusion.FindStringIndex(sourcePath)
 			if match != nil {
-				return nil
+				logger.Log.Debugf("Skipping %s", sourcePath)
```

Comment:
The log message seems a little too vague.

Reply:
Fixed.